### DISCUSSION TITLE
Trigger actions or set variables via CLI

### DISCUSSION THREAD

**Discussion by UberKitten** [File-New-Project/EarTrumpet#1083] [Ideas]
The actions in the dev branch look awesome!

There are several tasks I'm constantly using EarTrumpet for, like setting my default audio output to my headphones or a specific app to output to my TV audio.

I'd like to automate some of these actions from various sources, like Home Assistant, PowerShell, Scheduled Tasks, etc. For example, I have a button in Home Assistant that switches the TV input to my PC. I want to also run a command on my PC to change the audio device. I have an EarTrumpet action for that, so it'd be nice to just trigger that.

You can start EarTrumpet with CLI args [like this](https://superuser.com/a/1702711):
`start shell:appsFolder\40459File-New-Project.EarTrumpet_725pr5jq8wr8a!EarTrumpet "arguments here"`
(Worked in PowerShell for me but not cmd.exe, may take some fiddling)

It would be great if we could trigger actions from CLI:
`start shell:appsFolder\40459File-New-Project.EarTrumpet_725pr5jq8wr8a!EarTrumpet "--run My Action"`
And set variables from CLI:
`start shell:appsFolder\40459File-New-Project.EarTrumpet_725pr5jq8wr8a!EarTrumpet "--set MyVar true"`

I don't think you need to support many CLI arguments. There are other utilities like SoundVolumeView from Nirsoft that have very robust CLI support if someone is looking for extensive control. It would just be a nice feature for power users.

**Comment by Coldblackice** [comment]
An EarTrumpet CLI would be _fantastic_.
